app/src/tests/components/contents/runner/run/PacketRunForm.test.tsx

```diff
@@ -6,6 +6,9 @@ import { server } from "../../../../../msw/server";
 import { rest } from "msw";
 import { MemoryRouter, Route, Routes } from "react-router-dom";
 import { getTimeDifferenceToDisplay } from "../../../../../lib/time";
+import appConfig from "../../../../../config/appConfig";
+
+const testBaseUrl = "http://localhost";
```

Comment:
Would it make sense for `${testBaseUrl}${appConfig.apiUrl()}` to be defined as a common constant somewhere? It might even make more sense to define it as `new URL(appConfig.apiUrl(), testBaseUrl)` instead.

I don't really know much about the frontend tests, how/where is `window.location.href` defined in this case? Can we use that source of truth instead of defining it ourselves?

Reply:
I don't know where window.location gets defined - seems hard to find out! So I tried to find ways to set the location ourselves. I've given up trying to use a base other than 'localhost' in the test, but I _have_ found a way to mock the location's paths, which is helpful for testing that the current path doesn't get fed through to the url string.

Reply:
> It might even make more sense to define it as new URL(appConfig.apiUrl(), testBaseUrl) instead.

I think that would reduce my confidence in the tests - I want to test that, when we use this `URL` module, the resulting string is what we expect. So I want the util under test to handle the building of the URL, and I don't want to use any URL building utils in the test, just plain string interpolation.

Reply:
I've made the absolute api url into a common test helper, and the test base url into a common constant.